### PR TITLE
Lack of error handling for invalid address length

### PR DESCRIPTION
**Pull request by r8d8**
Refactored address to use byte array

### CODE REVIEW

File: pkg/keypair/keypair.go [hunk 1]
```diff
@@ -20,17 +20,27 @@ import (
...
-type Address string
+type Address [ADDRESS_BYTES]byte
 
 // KeyPair is a private/public crypto key pair.
 type KeyPair struct {
 	Private []byte
 	Public  []byte
 }
 
+func (addr Address) FromString(s string) (Address, error) {
```
Comment (dulanov) [inline pkg/keypair/keypair.go]: It's a constructor, you don't need here `(addr Address)`

File: pkg/keypair/keypair.go [hunk 2]
```diff
@@ -51,20 +61,20 @@ func (pair KeyPair) Address(chain core.Chain) Address {
 	b := append([]byte{chain.Id}, r.Sum(nil)...)
 	h = sha3.SumKeccak256(b)
 	a := append(b, h[:4]...)
-	return Address(base32.StdEncoding.EncodeToString(a))
+
+	addr := Address{}
+	copy(addr[:], a[:])
```
Comment (dulanov) [inline pkg/keypair/keypair.go]: Why not istead just `Address{a[:]}`?

Comment (r8d8) [reply pkg/keypair/keypair.go]: prohibited conversion, go doesn't allow

File: pkg/keypair/keypair.go (hunk 1 above)
Comment (r8d8) [reply pkg/keypair/keypair.go]: true, 
refactored accordingly

File: pkg/keypair/keypair.go [hunk 3]
```diff
@@ -31,7 +31,8 @@ type KeyPair struct {
 	Public  []byte
 }
 
-func (addr Address) FromString(s string) (Address, error) {
+func FromString(s string) (Address, error) {
+	addr := Address{}
```
Comment (dulanov) [inline pkg/keypair/keypair.go]: Better `var addr Address`

**Pull request merged**